src/include/sof/trace.h

```diff
@@ -186,7 +186,8 @@ typedef void(*log_func)();
 		log_function = (log_func)&function_name##2;	\
 	if (PP_NARG(__VA_ARGS__) == 3)				\
 		log_function = (log_func)&function_name##3;	\
-	log_function(entry, ##__VA_ARGS__);			\
+	if (log_function)					\
```

Comment:
It looks like switch() or if else() would be more optimal here. i.e. compiler will do less tests